impl/vfat/cogent/plat/linux/wrapper.ac

```diff
@@ -0,0 +1,1068 @@
+/*
+ * Copyright 2016, NICTA
+ *
+ * This software may be distributed and modified according to the terms of
+ * the GNU General Public License version 2. Note that NO WARRANTY is provided.
+ * See "LICENSE_GPLv2.txt" for details.
+ *
+ * @TAG(NICTA_GPL)
+ */
+
+$esc:(#include <adt.h>)
+$esc:(#include <wrapper.h>)
+$esc:(#include <fat.h>)
+
+$esc:(#include <wraphead.h>)
+
+typedef struct super_block *Superblock;
+typedef void *ExState;
+typedef struct inode *VfsInode;
+typedef void *ErrPtr;
+typedef struct dentry *VfsDentry;
+
+#include <generated.c>
+#include <abstract.h>
+#include <gum/anti/iterator.ac>
+
+/*=================*
+Extern these eventually
+*=================*/
+static inline void fat_lock_build_inode(struct msdos_sb_info *sbi)
+{
+	if (sbi->options.nfs == FAT_NFS_NOSTALE_RO)
+		mutex_lock(&sbi->nfs_build_inode_lock);
+}
+
+static inline void fat_unlock_build_inode(struct msdos_sb_info *sbi)
+{
+	if (sbi->options.nfs == FAT_NFS_NOSTALE_RO)
+		mutex_unlock(&sbi->nfs_build_inode_lock);
+}
+
+/*=================*
+Translators
+*=================*/
+// change previous translations to use this instead
+struct timespec * cog_to_c_time ($ty:(Time) arg){
+	struct timespec * ts = kmalloc(sizeof(struct timespec), GFP_KERNEL);
+	if(!ts){
+		$esc:(printk(KERN_WARNING "ERROR MALLOCING.\n")); //TODO: fix errors
```

Comment:
Compiler error when `$esc` is removed:

```
Glue code (functions) generation failed: 
Error: Failed to parse C: plat/linux/wrapper_pp.ac:2855:22:
    unclosed `('
Compilation failed!
```

Works without `$esc` if `KERN_WARNING` is removed, though.